components/multiselect/multiselect.tsx

```diff
@@ -1,115 +1,146 @@
-        numRemainingText: PropTypes.node,
-        buttonSubmitText: PropTypes.node,
-        buttonSubmitLoadingText: PropTypes.node,
-        submitImmediatelyOn: PropTypes.func,
-        saving: PropTypes.bool,
-        loading: PropTypes.bool,
-        placeholderText: PropTypes.string,
-    }
+export default class MultiSelect extends React.Component<Props, State> {
+    public listRef = React.createRef<MultiSelectList>()
+    public reactSelectRef = React.createRef<ReactSelect>()
+    public selected: Value | null = null
 
-    static defaultProps = {
+    public static defaultProps = {
         ariaLabelRenderer: defaultAriaLabelRenderer,
     }
 
-    constructor(props) {
+    public constructor(props: Props) {
         super(props);
 
-        this.selected = null;
-
         this.state = {
+            a11yActive: false,
             page: 0,
             input: '',
         };
     }
 
-    componentDidMount() {
-        document.addEventListener('keydown', this.handleEnterPress);
-        if (this.refs.reactSelect) {
-            this.refs.reactSelect.select.inputRef.addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-            this.refs.reactSelect.select.inputRef.addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
+    public componentDidMount() {
+        const inputRef: unknown = this.reactSelectRef.current && this.reactSelectRef.current.select.inputRef;
+
+        document.addEventListener<'keydown'>('keydown', this.handleEnterPress);
+        if (inputRef && typeof (inputRef as HTMLElement).addEventListener === 'function') {
+            (inputRef as HTMLElement).addEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
+            (inputRef as HTMLElement).addEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yActivateEvent);
 
-            this.refs.reactSelect.focus();
+            this.reactSelectRef.current!.focus(); // known from ternary definition of inputRef
         }
     }
 
-    componentWillUnmount() {
-        this.refs.reactSelect.select.inputRef.removeEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
-        this.refs.reactSelect.select.inputRef.removeEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
+    public componentWillUnmount() {
+        const inputRef: unknown = this.reactSelectRef.current && this.reactSelectRef.current.select.inputRef;
+
+        if (inputRef && typeof (inputRef as HTMLElement).addEventListener === 'function') {
+            (inputRef as HTMLElement).removeEventListener(A11yCustomEventTypes.ACTIVATE, this.handleA11yActivateEvent);
+            (inputRef as HTMLElement).removeEventListener(A11yCustomEventTypes.DEACTIVATE, this.handleA11yDeactivateEvent);
+        }
 
         document.removeEventListener('keydown', this.handleEnterPress);
     }
 
-    handleA11yActivateEvent = () => {
+    public handleA11yActivateEvent = () => {
```

Comment:
why all the handlers and internal methods are labeled as public? I'm not sure, but i think there is no need to make them public.